driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -834,15 +848,19 @@ public void onResult(@Nullable final ByteBuf result, @Nullable final Throwable t
         }
     }
 
-    private static final Logger COMMAND_PROTOCOL_LOGGER = Loggers.getLogger("protocol.command");
+    private static final StructuredLogger COMMAND_PROTOCOL_LOGGER = new StructuredLogger("protocol.command");
 
     private CommandEventSender createCommandEventSender(final CommandMessage message, final ByteBufferBsonOutput bsonOutput,
             final RequestContext requestContext) {
-        if (opened() && (commandListener != null || COMMAND_PROTOCOL_LOGGER.isDebugEnabled())) {
+        if (!isMonitoringConnection && opened() && (commandListener != null || COMMAND_PROTOCOL_LOGGER.isRequired(DEBUG, getClusterId()))) {
             return new LoggingCommandEventSender(SECURITY_SENSITIVE_COMMANDS, SECURITY_SENSITIVE_HELLO_COMMANDS, description,
                     commandListener, requestContext, message, bsonOutput, COMMAND_PROTOCOL_LOGGER);
         } else {
             return new NoOpCommandEventSender();
         }
     }
+
+    private ClusterId getClusterId() {
```

Comment:
Any chance of a NPE here?

Reply:
No there is always a ClusterId, just not always a description (which is null checked)